### ISSUE TITLE
Array.transpose does not calculate the size of the results correctly

### ISSUE THREAD

**Issue by prothSFDC** [paulroth3d/jupyter-ijavascript-utils#42]
```
arrayLen = (testArray) => (testArray || [])
  .reduce((max, line) => {
	const len = (line || []).length;
	return (len > max) ? len : max
  }, 0);

transpose = (matrix) => {
  //-- fast fail
  if (!matrix || !Array.isArray(matrix)) {
    return [];
  }

  //-- check for 1d arrays
  if (!Array.isArray(matrix[0])) {
    return matrix.map((v) => [v]);
  }

  //-- for speed, we use for loops.
  const rows = matrix.length
  const cols = arrayLen(matrix);
  let colI;
  let rowI;

  const result = Array.from(new Array(cols))
    .map((r) => Array.from(new Array(rows)));

  for (colI = 0; colI < cols; colI += 1) {
    for (rowI = 0; rowI < rows; rowI += 1) {
      result[colI][rowI] = matrix[rowI][colI];
    }
  }

  return result;
};
```

**Comment by paulroth3d** [comment]
ready for pull request / merge

**Comment by paulroth3d** [comment]
corrected in version 1.31
